hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -147,7 +147,7 @@ private void sanityCheck(Configuration conf, ResolvedSchema schema) {
         throw new HoodieValidationException("Option '" + FlinkOptions.PRECOMBINE_FIELD.key()
             + "' is required for payload class: " + DefaultHoodieRecordPayload.class.getName());
       }
-      if (preCombineField.equals(FlinkOptions.PRECOMBINE_FIELD.defaultValue())) {
+      if (preCombineField.equals(FlinkOptions.PRECOMBINE_FIELD.defaultValue()) || preCombineField.equals(FlinkOptions.NO_PRE_COMBINE)){
         conf.setString(FlinkOptions.PRECOMBINE_FIELD, FlinkOptions.NO_PRE_COMBINE);
```

Comment:
Is this change valid ? We still set up the option if it is already `FlinkOptions.NO_PRE_COMBINE`.

Reply:
it's valid, suppose a table with a 'ts' column, we want to set PRECOMBINE_FIELD=FlinkOptions.NO_PRE_COMBINE to disable the precombine field, above code will go into line 153, because preCombineField.equals(FlinkOptions.PRECOMBINE_FIELD.defaultValue()) is false, FlinkOptions.PRECOMBINE_FIELD.defaultValue() is 'ts', preCombineField is what we set 'no_precombine', you can also see the test.

Reply:
> Is this change valid ? We still set up the option if it is already `FlinkOptions.NO_PRE_COMBINE`.

@danny0405 looking forward to your reply

Reply:
One thing need to note here is that: option `FlinkOptions.NO_PRE_COMBINE` is only for internal use

Reply:
yes, i want the proc_time/natural order sequence semantics even though you got a ts field in the schema

Reply:
Then we can do such logic:
else if (!preCombineField.equals(FlinkOptions.NO_PRE_COMBINE)) {
    throw new HoodieValidationException("Field " + preCombineField + " does not exist in the table schema."
            + "Please check '" + FlinkOptions.PRECOMBINE_FIELD.key() + "' option.");
}

Reply:
yes, it is more elegant. I've modified to it